### PR TITLE
Feature/week8 homework

### PR DESCRIPTION
**Pull request by d-ben7**

### CODE REVIEW

File: Homework/Week8/app/src/main/java/com/android/pokedex/ui/HomeFragment.kt [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+import com.android.pokedex.viewmodel.PokemonViewModel
+import com.android.pokedex.viewmodel.SyncDataWorkerViewModel
+import kotlinx.coroutines.launch
+
+class HomeFragment : Fragment() {
+
+    private val adapter = PokemonRecyclerAdapter()
+    private lateinit var binding: FragmentHomeBinding
+    private lateinit var pokemonViewModel: PokemonViewModel
+    private lateinit var workerViewModel: SyncDataWorkerViewModel
```
Comment (filbabic) [inline Homework/Week8/app/src/main/java/com/android/pokedex/ui/HomeFragment.kt:26]: Avoid lateinit as much as you can. I mentioned this before in the previous assignment, and other sessions we've had.

Also, you don't need the worker here as a property, not do you need to observe its work.

You can start it, and let it update the DB, and you'll show the new data when the user goes back to the app!

File: Homework/Week8/app/src/main/java/com/android/pokedex/viewmodel/SyncDataWorkerViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import com.android.pokedex.worker.SyncDataWorker
+import java.util.concurrent.TimeUnit
+
+class SyncDataWorkerViewModel(application: Application) : AndroidViewModel(application) {
+
+    private val workManager = WorkManager.getInstance(application)
+    val workInfo: LiveData<WorkInfo>
+
+    init {
+        workInfo = scheduleSyncData()
```
Comment (filbabic) [inline Homework/Week8/app/src/main/java/com/android/pokedex/viewmodel/SyncDataWorkerViewModel.kt:16]: You can remove this, and let the worker jus do the sync, without exposing any data. It's even better that way, as your workers will mostly do things in the BG, and not while the user is in the app!

File: Homework/Week8/app/src/main/java/com/android/pokedex/worker/SyncDataWorker.kt [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+class SyncDataWorker(context: Context, params: WorkerParameters) :
+    CoroutineWorker(context, params) {
+
+    override suspend fun doWork(): Result {
+        try {
+            showWorkerNotification("Syncing data in progress", applicationContext)
+            val pokemonDao = PokemonDatabase.getInstance(applicationContext).pokemonDao
+
+            for (id in 1..39) {
```
Comment (filbabic) [inline Homework/Week8/app/src/main/java/com/android/pokedex/worker/SyncDataWorker.kt:18]: Should use this through the Repository, and not have the hardcoded logic here!

The notification is fine though!

**Comment by d-ben7** [comment]
Thanks for the all the feedback, Filip.

**Pull request merged**